actor/v2action/service_access.go

```diff
@@ -0,0 +1,125 @@
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		warnings, err := actor.CloudControllerClient.UpdateServicePlan(plan.GUID, true)
+		allWarnings = append(allWarnings, warnings...)
+		if err != nil {
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, nil
+}
+
+// EnablePlanForAllOrgs enables access to a specific plan of the given service in all orgs.
+func (actor Actor) EnablePlanForAllOrgs(serviceName, servicePlanName string) (Warnings, error) {
+	servicePlans, allWarnings, err := actor.GetServicePlansForService(serviceName)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	// We delete all service plan visibilities for the given Plan since the attribute public should function as a giant on/off
+	// switch for all orgs. Thus we need to clean up any visibilities laying around so that they don't carry over.
+	for _, plan := range servicePlans {
+		if plan.Name == servicePlanName {
+			warnings, err := actor.removeOrgLevelServicePlanVisibilities(plan.GUID)
+			allWarnings = append(allWarnings, warnings...)
+			if err != nil {
+				return allWarnings, err
+			}
+
+			ccv2Warnings, err := actor.CloudControllerClient.UpdateServicePlan(plan.GUID, true)
+			allWarnings = append(allWarnings, ccv2Warnings...)
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, actionerror.ServicePlanNotFoundError{PlanName: servicePlanName, ServiceName: serviceName}
+}
+
+// EnableServiceForOrg enables access for the given service in a specific org.
+func (actor Actor) EnableServiceForOrg(serviceName, orgName string) (Warnings, error) {
+	servicePlans, allWarnings, err := actor.GetServicePlansForService(serviceName)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	org, orgWarnings, err := actor.GetOrganizationByName(orgName)
+	allWarnings = append(allWarnings, orgWarnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		_, warnings, err := actor.CloudControllerClient.CreateServicePlanVisibility(plan.GUID, org.GUID)
+		allWarnings = append(allWarnings, warnings...)
+		if err != nil {
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, nil
+}
+
+// EnablePlanForOrg enables access to a specific plan of the given service in a specific org.
+func (actor Actor) EnablePlanForOrg(serviceName, servicePlanName, orgName string) (Warnings, error) {
+	servicePlans, allWarnings, err := actor.GetServicePlansForService(serviceName)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	org, orgWarnings, err := actor.GetOrganizationByName(orgName)
+	allWarnings = append(allWarnings, orgWarnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		if plan.Name == servicePlanName {
+			_, warnings, err := actor.CloudControllerClient.CreateServicePlanVisibility(plan.GUID, org.GUID)
+			allWarnings = append(allWarnings, warnings...)
+			return allWarnings, err
+		}
+	}
+
+	return nil, fmt.Errorf("Service plan '%s' not found", servicePlanName)
```

Comment:
This is an unfortunate error message that is different from line 52. This is legacy behaviour on the CLI which we'd love to fix. See below for examples of how to reproduce:

```
→ cf service-access
Getting service access as admin...
broker: overview-broker
   service            plan      access   orgs
   overview-broker    simple    all

→ cf enable-service-access overview-broker -p lol
Enabling access of plan lol for service overview-broker as admin...
FAILED
The plan lol could not be found for service overview-broker

→ cf enable-service-access overview-broker -p lol -o acceptance
Enabling access to plan lol of service overview-broker for org acceptance as admin...
FAILED
Service plan lol not found
```

The former message is more descriptive. Perhaps we could have a separate story to fix this up. @abbyachau

Reply:
Thanks for the explanation @williammartin. Abby is out this week, but I think it sounds reasonable to have a small follow-up story to fix this.

Reply:
Here is the follow-up story https://www.pivotaltracker.com/story/show/162148852

Reply:
@abbyachau let us know if you're happy for us to make this follow up change. We have blocked the story @georgi-lozev linked to on your response.

Reply:
hey @mattmcneeney @williammartin sorry I'm just seeing this because of the tracker ping (not getting the github notifications for these comments). I believe `Service plan lol not found` is the convention we try to follow for these types of error but I'm fine with changing it.